RecoLocalCalo/HcalRecAlgos/src/MahiFit.cc

```diff
@@ -91,7 +81,7 @@ void MahiFit::phase1Apply(const HBHEChannelInfo& channelData,
     //Dark current (for SiPMs)
     double noiseDC=0;
     if(channelData.hasTimeInfo() && !channelData.hasEffectivePedestals() && (charge-ped)>channelData.tsPedestalWidth(iTS)) {
-      noiseDC = darkCurrent;
+      noiseDC = getSiPMDarkCurrent(channelData.darkCurrent(),channelData.fcByPE(),channelData.lambda());
```

Comment:
as I noted on the other comment, this won't actually be used. Doing the computation for each channel costs time when effective pedestals are used for every scenario now. This could be done more efficiently (just compute the first time it's encountered in this loop and then reuse it), but again, it should never actually be called.

Reply:
> it should never actually be called

perhaps I'm missing the condition of when it can even be called.
Which parameter leads to this not being called?
It's not obvious what the compiler decides to do: unless there is a hint to not make this computation, it may actually still be done preemptively.
So, better clean up the code.

Reply:
primarily controlled by `channelData.hasEffectivePedestals()`, set in `HBHEPhase1Reconstructor` when its parameter `saveEffectivePedestal` is enabled (default for 2017-onward, which are the only scenarios with SiPMs - presence of SiPMs is indicated by `channelData.hasTimeInfo()`). This flag in the channel info means that the effective pedestals were accessed in the database.

Personally, I would rather just remove `getSiPMDarkCurrent()` from the code entirely.

Reply:
@kpedro88 @abdoulline I kept this getSiPMDarkCurrent for now

Can you check if for the 2017 GT the effective pedestal is exact copy of the qie pedestal "BIAS OFF" or we have the "BIAS ON" values loaded